src/router/index.js

```diff
@@ -0,0 +1,38 @@
+import { createWebHistory, createRouter } from 'vue-router';
+
+import landing from './landing';
+import auth from './auth';
+
+const routes = [
+  {
+    path: '/',
+    component: () => import('@/pages/LandingPanel.vue'),
+    children: [
+      ...landing,
+      {
+        path: '/404',
+        name: '404Page',
+        component: () => import('@/pages/404Page.vue'),
+      },
+    ],
+  },
+  {
+    path: '/',
+    component: () => import('@/pages/AuthPanel.vue'),
+    children: [...auth],
+  },
```

Comment:
Please tell me if this is a good practice. My goal was to create a path for authentication screens like
`/auth/login` and `/auth/register` (or just `/login` and `/register` like it is at the moment) where the component `AuthPanel.vue` is only accessible through one of those paths, and just `/auth` does not exist as an independent route.

Reply:
That's what I meant - it is repeated on purpose to be not accessible without children path. If I set this as `/auth` instead of repeated `/` I would have to create redirection logic, right? Because `/auth` as I planned should not be accessible, but `/auth/login` for example, should.

```
{
    path: '/',
    component: () => import('@/pages/AuthPanel.vue'),
    children: 
      {
        path: '/auth/login',  // did just `/login`
        name: 'LoginPage',
        component: () => import('@/pages/auth/LoginPage.vue'),
      },
      {
        path: '/auth/register', // did just `/register`
        name: 'RegisterPage',
        component: () => import('@/pages/auth/RegisterPage.vue'),
      }
}
```
Basically the goal is to make `/auth` not accessible at all, if I messed up my explanation 😅

Reply:
You could create additional route to handle that (additional child under the `/auth` route as `path: ''`) and redirect to, say `/auth/login` and it will be fine and work great 😄
Also... you could technically omit it - that's also okay and happens sometimes that we do not have the `something/` path but we have `something/else` path in our apps. Not ideal, yes, but it happens 😅

Reply:
This redirection sounds good anyway, so I'll go with the first approach 👍